### ISSUE TITLE
Pretrained checkpoints?

### ISSUE THREAD

**Issue by ex4sperans** [NeurAI-Lab/biHomE#10]
Hello guys! Awesome work - really liked reading the paper. It's very interesting that you've managed non-VGG backbones to work as a perceptual loss.

Unfortunately, I currently don't have compute resources to train the model from scratch, so would you mind sharing some pretrained checkpoints?

Thanks!

**Comment by dkoguciuk** [comment]
Hi Dmytro, 

Training all those models was time-consuming for sure. It's double hard because some of the architectures are hard to converge -- biHome also doesn't converge always, but it's much more likely than other models. Unfortunately, I don't have access to the trained checkpoints, I'm so sorry :disappointed:  

BTW. Is it really so uncommon to use a non-VGG network as perceptual loss? This is interesting! Maybe the difference here comes from a vastly different task than super-resolution? It would be nice to quantify this difference somehow and understand what's happening there and why.

Best,
Daniel

**Comment by ex4sperans** [comment]
Hi Daniel, 

Thanks for a quick response :) No worries about not being able to share the checkpoints.

Regarding VGG, it's been de-facto a standard perceptual loss for years. I remember many people trying other options and observing them not working as well as VGG. The argument has always been that VGG is very "inefficient" and unnecessarily big and thus it retains a lot of low-level information about the input image, while ResNets and other more optimized architectures tend to ignore the features that are not important for the original task (which is usually ImageNet classification). As far as I know, VGG-based perceptual loss is still widely used in tasks that require a per-pixel comparison, e.g. style transfer, superresolution, deblurring, etc.

One recent example is [Real-ESRGAN](https://openaccess.thecvf.com/content/ICCV2021W/AIM/html/Wang_Real-ESRGAN_Training_Real-World_Blind_Super-Resolution_With_Pure_Synthetic_Data_ICCVW_2021_paper.html). I just checked though, and there is a newer paper that improves the approach proposed in the Real-ESRGAN paper with a novel perceptual loss that combines both VGG and ResNet. 

Dmytro

**Comment by dkoguciuk** [comment]
Hi Dmytro,

hmm, I've googled for half an hour and didn't find anything other than VGG. However, in non-perceptual tasks, I expect ResNet will work better. Homography estimation is one example. I also thought about doing unsupervised local feature learning using perceptual loss, but then I changed the domain and never coded it.

For now, I will close the issue. Feel free to reopen if you'd like. Also, we can connect on some other medium ;) 

Best,
D
